schema/install_schema.go

```diff
@@ -18,7 +18,46 @@ type InstallSchema struct {
 	GrubOptionsSchema   `json:"grub_options,omitempty"`
 	Image               string `json:"image,omitempty" description:"Use a different container image for the installation"`
 	PowerManagement
-	SkipEncryptCopyPlugins bool `json:"skip_copy_kcrypt_plugin,omitempty"`
+	SkipEncryptCopyPlugins bool                `json:"skip_copy_kcrypt_plugin,omitempty"`
+	Partitions             ElementalPartitions `json:"partitions,omitempty" mapstructure:"partitions"`
+	Firmware               string              `json:"firmware,omitempty" mapstructure:"firmware"`
+	PartTable              string              `json:"part-table,omitempty" mapstructure:"part-table"`
+	GrubDefEntry           string              `json:"grub-entry-name,omitempty" mapstructure:"grub-entry-name"`
+	ExtraPartitions        []*Partition        `json:"extra-partitions,omitempty" mapstructure:"extra-partitions"`
+	Force                  bool                `json:"force,omitempty" mapstructure:"force"`
+	ExtraDirsRootfs        []string            `json:"extra-dirs-rootfs,omitempty" mapstructure:"extra-dirs-rootfs"`
+	Active                 Image               `json:"system,omitempty" mapstructure:"system"`
+	Recovery               Image               `json:"recovery-system,omitempty" mapstructure:"recovery-system"`
+	Passive                Image               `json:"passive,omitempty" mapstructure:"recovery-system"`
+}
+
+// ImageSource represents the source from where an image is created for easy identification
+type ImageSource struct {
```

Comment:
do we need this here? I dont think it will work, i.e. you cannot override the Source by a yaml directly like this, you need to override the `install.Source` like here:

```
func generateInstallConfForCLIArgs(sourceImageURL string) string {
	if sourceImageURL == "" {
		return ""
	}

	return fmt.Sprintf(`install:
  source: %s
`, sourceImageURL)
}
```

then that yaml gets ingested and converted to an actual ImageSource which is for internal use mainly

Reply:
so what should source be? just a string? or it shouldn't be there at all?

Reply:
made it a string, now, let me know if it's not correct